akka-actor/src/main/java/akka/dispatch/AbstractBoundedNodeQueue.java

```diff
@@ -167,9 +167,17 @@ public final T poll() {
             final Node<T> deq = getDeq();
             final Node<T> next = deq.next();
             if (next != null) {
+                //Can't rely on next.value after casDeq because 'next' can be popped by another thread already
+                //before this one finishes. 
```

Comment:
This is not true: only one thread can win the CAS operation, and that one then ownes `next`. The loser goes back to the outer loop and reads the new `next` pointer from the new `deq` that it obtains (because we just shifted it).